server/services/userService.ts

```diff
@@ -9,12 +9,37 @@ import type { SystemToken } from '@hmpps-auth'
 import type { User } from '@manage-users-api'
 import type { Caseload } from '@prison-api'
 
+const throwUserLoadNon404Error = (username: string, knownError: ResponseError) => {
+  const errorMessage =
+    knownError.message === 'Internal Server Error' ? `Error fetching user ${username}.` : knownError.message
+  throw createError(knownError.status || 500, errorMessage)
+}
+
 export default class UserService {
   constructor(
     private readonly hmppsManageUsersClientBuilder: RestClientBuilder<HmppsManageUsersClient>,
     private readonly prisonApiClientBuilder: RestClientBuilder<PrisonApiClient>,
   ) {}
 
+  /*
+   * Rather than return a boolean this function returns a thruthy by way of the User object
+   * This likely saves on a subsequent call afterwards to fetch these details.
+   */
+  async checkUserExistsFromUsername(
```

Comment:
I'm not sure I understand what this is used for - are we making a subsequent call after this (re: the "saves on subsequent call afterwards")?

Would combining the two functions into one work?

```
 async getUserNameOrUnknown(userToken: Express.User['token'], username: User['username']): Promise<string> {
    const hmppsManageUsersClient = this.hmppsManageUsersClientBuilder(userToken)

    let user: User | null

    try {
      user = await hmppsManageUsersClient.getUserFromUsername(username)
    } catch (error) {
      const knownError = error as ResponseError
      user = knownError.status === 404 ? null : throwUserLoadNon404Error(username, knownError)
    }
    return user ? StringUtils.convertToTitleCase(user.name) : `User '${username}' not found`
  }
```

Reply:
OK, I'd like us not to keep going back and forth on this. To get this over the line, would you make `checkUserExistsFromUsername` method private, so it's clear to devs it's not used outside this class. That'll be enough to get this PR in.

Then (separate piece of work, I'll write a ticket) tidy up `getUserFromUsername` and this method so we don't have these duplicate methods hanging around (No 2 in the blog post you linked is "delete duplicate methods" after all).